components/KBar/Button.tsx

```diff
@@ -6,6 +8,15 @@ import { Button } from 'components/ui'
 
 const KBarButton = () => {
   const { query } = useKBar()
+  const [isMacUser, setIsMacUser] = useState(false)
```

Comment:
Why do you need to add an internal state to this component ? Is it not possible to use `isMac` directly as it was done before ?

Reply:
The isMac() utility uses the user-agent to determine if user is on mac or not. This functionality doesn't work with SSR because it doesn't have access to user-agent so the existing code was always server-side rendering with isMac = false causing an error because there was a mismatch between the client rendered-code and the server-rendered code. Evm.codes doesn't have this issue because they do not use Next but React. In our case we need to put that logic in a state to avoid this rendering mismatch

Reply:
Oh ok, thanks for the detailed explanation 👍